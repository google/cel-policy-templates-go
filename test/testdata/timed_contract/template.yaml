# Copyright 2019 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

apiVersion: policy.acme.co/v1
kind: PolicyTemplate
metadata:
  name: timed_contract
  namespace: acme
schema:
  type: object
  properties:
    description:
      type: string
    resource_prefix:
      type: string
    start:
      type: string
      format: date-time
    end:
      type: string
      format: date-time
evaluator:
  productions:
    - match: >
        resource.name.startsWith(rule.resource_prefix) &&
        (request.time < rule.start || request.time > rule.end)
      decision: policy.deny
      output: true
