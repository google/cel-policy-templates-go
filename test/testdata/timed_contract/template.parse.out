1~# Copyright 2019 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

2~apiVersion: 3~"policy.acme.co/v1"
4~kind: 5~"PolicyTemplate"
6~metadata:7~
  8~name: 9~"timed_contract"
  10~namespace: 11~"acme"
12~schema:13~
  14~type: 15~"object"
  16~properties:17~
    18~description:19~
      20~type: 21~"string"
    22~resource_prefix:23~
      24~type: 25~"string"
    26~start:27~
      28~type: 29~"string"
      30~format: 31~"date-time"
    32~end:33~
      34~type: 35~"string"
      36~format: 37~"date-time"
38~evaluator:39~
  40~productions:41~
    - 42~43~match: 44~>
        resource.name.startsWith(rule.resource_prefix) &&
        (request.time < rule.start || request.time > rule.end)
      45~decision: 46~"policy.deny"
      47~output: 48~true
